app/dao/api_key_dao.py

```diff
@@ -29,6 +30,12 @@ def expire_api_key(service_id, api_key_id):
     db.session.add(api_key)
 
 
+def get_api_key_by_secret(secret):
+    return ApiKey.query.filter_by(
+        _secret=encryption.encrypt(str(secret))
```

Comment:
The `secret` column already has a unique index so this will be a fast lookup in the database